### PR TITLE
Change current dir for go get swagger so it doesn't modify go.mod

### PR DESCRIPTION
**Pull request by guillep2k**
This PR fixes the CI vendor problem by keeping `go get` from changing `go.mod` every time it downloads swagger. In a long conversation with @mrsdizzie and @techknowlogick it was established that this seems to be the culprit of the latest unintended vendor changes:

```
Makefile:
		GO111MODULE="on" $(GO) get -u github.com/go-swagger/go-swagger/cmd/swagger@v0.20.1; \
```

That produces changes in gitea's `go.mod` every time a new version of a golang library is released.

To work around that problem I've come with this solution, which is not perfect but could work until a better one comes up.

It works by executing `$(GO) get` outside gitea's directory, so it won't see/modify `go.mod`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -102,7 +102,9 @@ generate:
 .PHONY: generate-swagger
 generate-swagger:
 	@hash swagger > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
+		cd .. ; \
```
Comment (typeless) [inline Makefile:105]: According to `go help modules`
```
If invoked with -mod=readonly, the go command is disallowed from the implicit
automatic updating of go.mod described above. Instead, it fails when any changes
to go.mod are needed. This setting is most useful to check that go.mod does
not need updates, such as in a continuous integration and testing system.
The "go get" command remains permitted to update go.mod even with -mod=readonly,
and the "go mod" commands do not take the -mod flag (or any other build flags).
```
I wonder if `-mod=readonly` could be helpful?

Comment (lafriks) [reply Makefile:105]: it will make go get fail but we just need that it would not update go.mod

Comment (typeless) [reply Makefile:105]: Yeah, you're right. 🤦‍♂

File: Makefile [hunk 2]
```diff
@@ -102,7 +102,9 @@ generate:
...
 		GO111MODULE="on" $(GO) get -u github.com/go-swagger/go-swagger/cmd/swagger@v0.20.1; \
+		cd gitea ; \
```
Comment (lafriks) [inline Makefile]:
```suggestion
		cd - ; \
```

File: Makefile [hunk 3]
```diff
@@ -120,7 +122,9 @@ swagger-check: generate-swagger
 .PHONY: swagger-validate
 swagger-validate:
 	@hash swagger > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
+		cd .. ; \
 		GO111MODULE="on" $(GO) get -u github.com/go-swagger/go-swagger/cmd/swagger@v0.20.1; \
+		cd gitea ; \
```
Comment (lafriks) [inline Makefile]:
```suggestion
		cd - ; \
```

**Review by lafriks — CHANGES_REQUESTED** [review]

**Comment by typeless** [comment]
I am a bit confused about the issue. If the we already specify the exact version of go-swagger, wouldn't it just update the `go.mod` once if we have committed the change?

File: Makefile (hunk 2 above)
Comment (guillep2k) [reply Makefile]: @lafriks I think that might not work with some limited versions of shell. Currently Linux's `/bin/sh` seems to handle it, but I do remember in the old days AIX or Solaris woudn't like it. But may be I'm too old.

Comment (lafriks) [reply Makefile]: It's still better to have such way as otherwise what if I have cloned project in other directory than `gitea`?

**Comment by guillep2k** [comment]
> 
> 
> I am a bit confused about the issue. If the we already specify the exact version of go-swagger, wouldn't it just update the `go.mod` once if we have committed the change?

You have a point, but we're not using a binary release from go-swagger. It may be pointing to `master` in some of its dependencies.

**Comment by mrsdizzie** [comment]
See https://github.com/go-gitea/gitea/issues/8079 for an explanation of why current behavior causes these type of problems and should get a proper fix going forward

**Comment by typeless** [comment]
@guillep2k That sounds like a headache if a dependency could potentially have their dependencies inconstant once the semantic version has been specified. I expected that the MVS rules of Go modules do not work that way.

**Comment by guillep2k** [comment]
> 
> 
> @guillep2k That sounds like a headache if a dependency could potentially have their dependencies inconstant once the semantic version has been specified. I expected that the MVS rules of Go modules do not work that way.

@typeless You are right again. @mrsdizzie explained it better in #8079.

**Review by lafriks — APPROVED** [review]

**Comment by lunny** [comment]
We just need go-swagger as a command line tool but not depends that on gitea runtime. `go get` have these two feature. download a library or a tool. I think where the problem is.

**Review by lunny — APPROVED** [review]

**Comment by lafriks** [comment]
Closing in favor of #8087

**Pull request closed without merging**